### DISCUSSION TITLE
Enable export

### DISCUSSION THREAD

**Discussion by Azzerty23** [icflorescu/mantine-datatable#122] [Q&A]
Is it possible to add an export feature ?
I imagine to have an `exportable` property in DataTableColumn (like `sortable`) and a `DataTableExportProps` to define filename and supported file type (csv and/or Excel)

**Comment by icflorescu** [comment]
I think this kind of falls outside the scope of the project.

Think about the implications:
There's a potentially infinite number of ways to export data; different people will have different needs and opinions and there's simply no generic way of handling this.
Some would want their entire dataset to be exported (even if the table is showing a page).
There's also the problem of correctly formatting/localizing the exported data for Excel.
For instance, in my projects I've been using this feature with `exceljs` (an excellent package I highly recommend), which is not exactly trivial to use and requires heavy customization if you need useful output data.

So, to cut a long story short, I'm of the opinion that you should implement your own export mechanism in your projects.

**Reply by Azzerty23** [reply]
I see, thank you for your quick reply and for all the work !

**Reply by icflorescu** [reply]
Thank you for your kind words!

**Reply by otobot1** [reply]
Yeah I agree. This would add a lot of bloat to the datatable code and is better handled in a parent component (using whatever solution works for the use case).
